File: BE-Task_1-Go/internal/repository/memory/store.go
package memory

import (
	"sync"

	"gotest/internal/domain/entities"
)

type Store struct {
	mu        sync.RWMutex
	agents    map[string]entities.Agent
	purchases map[string]entities.Purchase
	refunds   map[string]entities.Refund
}

func NewStore() *Store {
	return &Store{
		agents:    make(map[string]entities.Agent),
		purchases: make(map[string]entities.Purchase),
		refunds:   make(map[string]entities.Refund),
	}
}

File: BE-Task_1-Php/internal/repository/memory/store.php
<?php

declare(strict_types=1);

namespace App\Internal\Repository\Memory;

use App\Internal\Domain\Entities\Agent;
use App\Internal\Domain\Entities\Purchase;
use App\Internal\Domain\Entities\Refund;

final class Store
{
    /** @var array<string, Agent> */
    public array $agents = [];

    /** @var array<string, Purchase> */
    public array $purchases = [];

    /** @var array<string, Refund> */
    public array $refunds = [];

    /**
     * @param array<string, Agent> $agents
     * @param array<string, Purchase> $purchases
     * @param array<string, Refund> $refunds
     */
    public function reset(array $agents, array $purchases, array $refunds): void
    {
        $this->agents = $agents;
        $this->purchases = $purchases;
        $this->refunds = $refunds;
    }
}
